### DISCUSSION TITLE
Any way to run with mistral API?

### DISCUSSION THREAD

**Discussion by nekopep** [bionic-gpt/bionic-gpt#425] [Q&A]
Hi, I'm exploring the capabilities of bionic-gpt and would like to use it as a gateway for several people in the same company with different roles (HR/Sales/R&D). So the team function sounds interesting, and I'd like to connect it with the mistral AI API (for confidentiality issues and to support European companies :) )
The idea is for the beginning, to explore the AI capabilities, so we prefer to use mistral compute power.
The problem is that mistral provides an endpoint and an API key, but we also need to be able to choose the model we're interested in.(https://api.mistral.ai/v1/models).
So far I managed to install bionic GPT on a debian KVM with 10 cores/ 32GB of RAM. (even if I'm not really used to K8s) 
Sadly I can't manage to find a setting to make Mistral endpoint working.
![image](https://github.com/bionic-gpt/bionic-gpt/assets/2825997/92800170-d718-48d5-902d-308ebe469c9a)

![image](https://github.com/bionic-gpt/bionic-gpt/assets/2825997/43d5504d-4e62-426d-a527-6ef8e21a7106)

Do you have any recommendation? Is the docker way also supported?

**Comment by ianpurton** [comment]
Hi, 

When you set the model name in the first screenshot, that's what gets passed by the API.

So the model name will need to match the name of the models at mistral.

Then you'll need a model config for each model you want to use.

Does that make sense?

**Reply by nekopep** [reply]
OK will give it a try, sure it was probably the issue.

**Comment by RedyJack** [comment]
When I have been playing with this I used ollama in a docker and defined mixtral like so in the bionicGPT.
![image](https://github.com/bionic-gpt/bionic-gpt/assets/148439298/7ee0b857-cee2-4be7-baea-b0080fc5a1ed)
Just remember to connect docker networks if running bionic in another container.

**Comment by ianpurton** [comment]
Did you follow these instructions? https://bionic-gpt.com/docs/community-edition/ollama/

**Reply by nekopep** [reply]
Nope, My plan was to directly use mistral API with my API key, but will test with the right Model Name. I as thinking the Model Name was the "Display Name". Will try tonight.

**Reply by ianpurton** [reply]
Oh sorry. Misunderstood.

Yes, thats right, model name is also the name passed down to the API.
